### DISCUSSION TITLE
Multiple s3 sources?

### DISCUSSION THREAD

**Discussion by Hurlyburly2** [cantaloupe-project/cantaloupe#573] [Q&A] [answered]
Hello,

We are running an instance of cantaloupe to serve images from multiple aws s3 buckets. As of now, all buckets have used the same endpoint, region and access/secret keys. The specific bucket is picked dynamically in the `s3source_object_info` method in delegates.rb.

We now want to serve images from buckets where those aforementioned values can differ. Is it possible to change these values on the fly, or would multiple cantaloupe instances be our only option?

Thanks!

**Comment by adolski** [comment, marked as answer]
Hi,

You should be able to do this on the `develop` branch (6.0-in-progress), but in the 5.x series, multiple instances is the only option.

**Reply by Hurlyburly2** [reply]
Thank you @adolski ! We are excited to take advantage of that feature, is there an estimated planned release time for 6.0, or is that as yet undecided?

**Reply by adolski** [reply]
It's a long way off, unfortunately.
